Documentation/Blazorise.Docs/Routes.razor.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Blazored.LocalStorage;
+using Blazorise.Docs.Services;
+using Blazorise.Modules;
+using Microsoft.AspNetCore.Components;
+
+namespace Blazorise.Docs;
+
+public partial class Routes
+{
+    [Inject] private ThemeService ThemeService { get; set; }
+    [Inject] private IJSUtilitiesModule JSUtilitiesModule { get; set; }
+    [Inject] ILocalStorageService LocalStorage { get; set; }
+
+    protected override void OnInitialized()
+    {
+        ThemeService.ThemeChanged += OnThemeChanged;
+
+        base.OnInitialized();
+    }
+
+    protected override async void OnAfterRender( bool firstRender )
+    {
+        if ( firstRender )
+        {
+            var theme = await LocalStorage.GetItemAsync<string>( "theme" );
+
+            if ( ThemeService.CurrentTheme != theme )
```

Comment:
What about putting the check (`( ThemeService.CurrentTheme != theme )`) inside the `SetTheme`?
This will also "protect" from the needless re-renders when user with light theme clicks again the light theme..